src/genie/libs/parser/iosxe/show_bgp.py

```diff
@@ -4815,20 +4808,40 @@ def cli(self, neighbor, address_family='', output=None):
                             '( +\(default for vrf +(?P<default_vrf>(\S+))\))?'
                             '( +VRF Router ID (?P<vrf_router_id>(\S+)))?$')
 
-        # Get VRF name by executing 'show bgp all neighbors | i BGP neighbor'
-        out_vrf = self.device.execute('show bgp all neighbors | i BGP neighbor')
         vrf = 'default'
-        for line in out_vrf.splitlines():
-            line = line.strip()
-            # BGP neighbor is 10.16.2.2,  remote AS 100, internal link
-            m = p.match(line)
-            if m:
-                if m.groupdict()['bgp_neighbor'] == neighbor:
-                    if m.groupdict()['vrf']:
-                        vrf = str(m.groupdict()['vrf'])
-                        break
-                else:
-                    continue
+        if output is None:
+            # Get VRF name by executing 'show bgp all neighbors | i BGP neighbor'
+            out_vrf = self.device.execute('show bgp all neighbors | i BGP neighbor')
+            for line in out_vrf.splitlines():
+                line = line.strip()
+                # BGP neighbor is 10.16.2.2,  remote AS 100, internal link
+                # BGP neighbor is 10.225.10.253,  vrf CE1test,  remote AS 60000, external link
+                # BGP neighbor is 192.168.0.254,  vrf L3VPN_1001,  remote AS 60001, external link
+                bgp_neighbor_re = re.compile(r'^BGP +neighbor +is +(?P<bgp_neighbor>[0-9A-Z\:\.]+)'
+                                '(, +vrf +(?P<vrf>\S+))?, +remote AS '
+                                '+(?P<remote_as_id>[0-9]+), '
+                                '+(?P<internal_external_link>[a-z\s]+)$')
+                m = bgp_neighbor_re.match(line)
+                if m:
+                    if m['bgp_neighbor'] == neighbor:
+                        if m['vrf']:
+                            vrf = str(m['vrf'])
+                            break
+                    else:
+                        continue
+        else:
+            if self.check_number_of_prefixes(output) == 0:
+                return {
+                    "vrf": {
+                        "default": {
+                            "neighbor": {
+                                neighbor: {
+                                    "address_family": {}
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
This is considered not great practice as far as genieparser goes. Would likely be better to remove this

Reply:
I don't like it either but if number of prefixes is 0, genieparser is not parsing the command output, being:

```Total number of prefixes: 0```

Maybe I need to impact another file?

Reply:
If that's the case, then the next best option would be to return `{}` instead of that pre-made dict. That way a `SchemaEmptyError` would be raised which would be more useful than a dict with no information in it